### DISCUSSION TITLE
How to find historical versions of the model?

### DISCUSSION THREAD

**Discussion by Hikari1118** [Open-Systems-Pharmacology/Forum#1784] [Questions & Problems]
How to find historical versions of the model?
I just made a mistake during the operation and deleted a very important and complex structure of the model, and then accidentally pressed the save button. What can I do now to retrieve the lost data?

**Comment by PavelBal** [comment]
You can try to "undo" your recent actions from the history. See this section of the documentation about the rollback function: https://docs.open-systems-pharmacology.org/shared-tools-and-example-workflows/history-manager-history-reporting
